### ISSUE TITLE
Error in dice loss

### ISSUE THREAD

**Issue by weinman** [argman/EAST#288]
I believe there's an error in the calculation of the denominator of the dice loss in this code.

The [Milletari et al.](https://arxiv.org/pdf/1606.04797.pdf) paper introducing the Dice coefficient as a loss function uses the formula

```
        ∑ᵢ pg
D =  ----------
      ∑ᵢp² + ∑ᵢg²
```

But [this line in argman/EAST/model.py](https://github.com/argman/EAST/blob/dca414de39a3a4915a019c9a02c1832a31cdd0ca/model.py#L96) neglects to square the term in the denominator. (The multiplication by `training_mask` simply serves to eliminate terms from the sum).

Instead, the line should read:

```python
union = tf.reduce_sum(tf.square(y_true_cls) * training_mask) + tf.reduce_sum(tf.square(y_pred_cls) * training_mask) + eps
```

I find performance does improve slightly when I use this corrected version.

**Comment by xulingzhi951118** [comment]
In the paper EAST, the loss function used for ROBX is IOU loss, not the Dice loss.
![image](https://user-images.githubusercontent.com/29685301/61638417-f0cfce80-accb-11e9-9956-5877b7f44450.png)

**Comment by weinman** [comment]
@xulingzhi951118 It's true of course that the original paper uses the IoU loss for RBOX *geometry*, but that is true of this repository as well (see the definition of [L_AABB](https://github.com/argman/EAST/blob/dca414de39a3a4915a019c9a02c1832a31cdd0ca/model.py#L130) in [model.py](https://github.com/argman/EAST/blob/dca414de39a3a4915a019c9a02c1832a31cdd0ca/model.py).

However, that's not germane to the issue here, as I am pointing out a discrepancy between what this code presents as Dice loss for the *classification* score, and how that loss should be calculated. 

To be completely clear, the original EAST paper uses a balanced cross-entropy loss; the  code in this repository purporting to be Dice loss does not match the definition given by Milletari et al.

**Comment by xulingzhi951118** [comment]
Oh, you are right. The calculation of dice loss really lack the square operation.
